### PR TITLE
feat(su_modal): Reload after su modal closes in flow where user is restricted

### PR DESCRIPTION
**Pull request by maxiuyuan**
Reload page when superuser modal is closed only during privileged access while not having permission. Based off of this comment

Additional super user modal feedback - typically the support team navigates directly to a view that requires the access (whereas in Richards’s example today the modal was brought up, authentication happened and then he navigated away). Is it possible that submitting the form will trigger the navigation action again? When in this view:

### CODE REVIEW

File: static/app/components/modals/sudoModal.tsx [hunk 1]
```diff
@@ -28,6 +28,7 @@ type Props = WithRouterProps &
      * User is a superuser without an active su session
      */
     isSuperuser?: boolean;
+    needsReload?: boolean;
```
Comment (evanpurkhiser) [inline static/app/components/modals/sudoModal.tsx:31]: Shouldn't sudo maybe just _always_ reload? Why is it just in this one place?

Is there maybe a way we can totally avoid a reload and re-init some client context?

Comment (maxiuyuan) [reply static/app/components/modals/sudoModal.tsx:31]: The purpose of this is for the flow where you access a customer's org but don't have superuser access. In this case, even after being granted superuser, you still will be on the same page with nothing loaded and an error message so a page refresh is probably the best way for it to load everything that you need.

In every other case of opening the superuse/sudo modal, you don't need to reload anything as the purpose is to be given superuser/sudo access and closing the modal after success is good

Comment (nhsiehgit) [reply static/app/components/modals/sudoModal.tsx:31]: Is there any case where they open the modal and are _not_ granted superuser access?

What would happen in that case after closing the modal?

**Review by nhsiehgit — APPROVED** [review]
functional changes lgtm

File: static/app/components/modals/sudoModal.tsx (hunk 1 above)
Comment (maxiuyuan) [reply static/app/components/modals/sudoModal.tsx:31]: That would only happen if the API request failed. That will be handled by `handleError` while this is in `handleSuccess`

**Pull request merged**